extensions/omni_sql/omni_sql.c

```diff
@@ -583,19 +583,129 @@ Datum is_returning_statement(PG_FUNCTION_ARGS) {
 
 PG_FUNCTION_INFO_V1(execute_parameterized);
 
+/** Holds information between multiple calls to the SRF execute_parameterized function */
+typedef struct ExecCtx {
+  char *stmt;
+  // returning is important to see if a given statement returns tuples
+  // to the caller. This controls the flow of the main function
+  bool returning;
```

Comment:
I think while we're re-organizing the code, it might make sense to document the fields here. Not a big deal, though.